Build fails with ncurses 6.0

After patching the Makefile to use `ncursesw6-config`, I get the following output from `make debug` (building using the snb-git AUR package):

```
cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong --param=ssp-buffer-size=4 -std=c99 -Wall -Werror -Wno-implicit-function-declaration -fstack-protector-all -fPIC -fPIE -DDEBUG -g  -D_GNU_SOURCE -c src/data.c -o src/data.o
cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong --param=ssp-buffer-size=4 -std=c99 -Wall -Werror -Wno-implicit-function-declaration -fstack-protector-all -fPIC -fPIE -DDEBUG -g  -D_GNU_SOURCE -c src/ui.c -o src/ui.o
cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong --param=ssp-buffer-size=4 -std=c99 -Wall -Werror -Wno-implicit-function-declaration -fstack-protector-all -fPIC -fPIE -DDEBUG -g  -D_GNU_SOURCE -c src/colors.c -o src/colors.o
src/data.c:10:0: error: "_GNU_SOURCE" redefined [-Werror]
 #define _GNU_SOURCE
 ^
<command-line>:0:0: note: this is the location of the previous definition
src/colors.c:5:0: error: "_GNU_SOURCE" redefined [-Werror]
 #define _GNU_SOURCE
 ^
src/ui.c:5:0: error: "_GNU_SOURCE" redefined [-Werror]
 #define _GNU_SOURCE
 ^
<command-line>:0:0: note: this is the location of the previous definition
<command-line>:0:0: note: this is the location of the previous definition
cc1: all warnings being treated as errors
Makefile:68: recipe for target 'src/colors.o' failed
make: *** [src/colors.o] Error 1
make: *** Waiting for unfinished jobs....
cc1: all warnings being treated as errors
Makefile:68: recipe for target 'src/data.o' failed
make: *** [src/data.o] Error 1
cc1: all warnings being treated as errors
Makefile:68: recipe for target 'src/ui.o' failed
make: *** [src/ui.o] Error 1
```

--- a/src/colors.c
+++ b/src/colors.c
@@ -2,7 +2,9 @@
  * Colours definition and setup function
  */
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 
 #include <ncurses.h>
 

--- a/src/data.c
+++ b/src/data.c
@@ -7,7 +7,9 @@
  * Also implements means of returning error information.
  */
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 
 #include <stdarg.h>
 #include <stdbool.h>

--- a/src/snb.c
+++ b/src/snb.c
@@ -1,4 +1,6 @@
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 #define _XOPEN_SOURCE_EXTENDED 1
 
 #include <stdbool.h>

--- a/src/ui.c
+++ b/src/ui.c
@@ -2,7 +2,9 @@
  * Whole UI code in one place
  */
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 
 #include <stdio.h>
 #include <stdbool.h>

